src/main/java/org/candlepin/subscriptions/tally/filler/ReportFiller.java

```diff
@@ -90,27 +97,33 @@ public void fillGaps(TallyReport report, OffsetDateTime start, OffsetDateTime en
 
             if (pending.isPresent() && lastSnapDate.isAfter(pendingSnapDate.get())) {
                 result.add(pending.get());
+                lastSnap = pending.get();
                 pending = Optional.empty();
                 pendingSnapDate = Optional.empty();
             }
 
             // Fill report up until the next snapshot, then add the snapshot to the report list.
-            result.addAll(fillWithRange(nextDate, lastSnapDate.minus(offset), offset));
+            result.addAll(fillWithRange(nextDate, lastSnapDate.minus(offset), offset, lastSnap,
+                useRunningTotalFormat));
             if (!pending.isPresent() || snapshotIsLarger(pending.get(), snapshot)) {
                 pending = Optional.of(snapshot);
                 pendingSnapDate = Optional.of(lastSnapDate);
             }
             nextDate = lastSnapDate.plus(offset);
         }
-        pending.ifPresent(result::add);
+        if (pending.isPresent()) {
+            result.add(pending.get());
+            lastSnap = pending.get();
+        }
 
         // If no snaps contain dates, just use the start of the range. Otherwise,
         // fill from the date of the last snapshot found, to the end of the range.
         if (lastSnapDate == null) {
-            result.addAll(fillWithRange(firstDate, lastDate, offset));
+            result.addAll(fillWithRange(firstDate, lastDate, offset, null, useRunningTotalFormat));
```

Comment:
What do you think of making overloaded `fillWithRange`, so there's not this "magic ~number~ null" parameter when reading through this logic?  Something like `fillWithRange(date,date,offset)` which will pass "null" and "false" as the last two parameters of `fillWithRange(date, date, offset, lastSnap, useRunningTotalFormat)`